sage_modabvar/abvar.py

```diff
@@ -635,7 +640,7 @@ def elliptic_curve(self):
         isogeny_classes = c.isogeny_classes(N)
         curves = [EllipticCurve(x[0][0]) for x in isogeny_classes]
 
-        return [E for E in curves if isogeny_check(E)][0]
+        return next(E for E in curves if isogeny_check(E))
```

Comment:
This is the worst possible algorithm to solve this problem -- you're making things much harder than necessary.  By the modularity theorem we know for a fact that there *will* be at least one curve that matches.  So instead of computing up to the sturm bound for f in the first place, which for at all large level could be very hard, you should instead just compute a few coefficients of f.  Then compare those few with every curve in curves.   If there is exactly one match, you're done.  If there is more than one match, repeat with more coefficients.    Often I bet only 3 or 4 a_p suffice.  Often for prime level you'll have `len(curves)=1`, in which case **zero** a_p are enough!